### PR TITLE
feat(path): add viewbox scaling

### PR DESCRIPTION
**Pull request by brycedorn**
hi! 👋 love the lib @ConnorAtherton - makes for some really cool animations of existing svg's with minimal effort.

one issue that I came across was that when using a svg with a predefined viewbox & resizing larger than that viewbox, the path length remained at the viewbox scale leading to incomplete drawings. to remedy this, I added a check for a viewbox definition in the the `Path` constructor, which then can get this scale & use it to calculate length.

let me know if this looks ok! I didn't test any line/polylines so only added it to path for now. also bumped the version & rebuilt (still on 0.0.6 on [npm](https://www.npmjs.com/package/walkway.js), would be great if you could bump it afterward 🙂).

### CODE REVIEW

File: src/walkway.js [hunk 1]
```diff
@@ -321,7 +321,11 @@
   function Path(path, duration, easing) {
     WalkwayElement.call(this, path, duration, easing);
 
-    this.length = path.getTotalLength();
+    var svg = path.closest('svg');
```
Comment (ConnorAtherton) [inline src/walkway.js]: Perhaps we could pull this out into a function -- `getElementScale` -- and put the same code in the Line and Polyline constructors. What do you think?

**Comment by ConnorAtherton** [comment]
@brycedorn Thanks for the contribution. I remember someone mentioning this to me in passing a few months after I first released this, but I forgot all about it until seeing this. Let me know what you think of the comment I added. 

Also, oversight on 0.0.7 not being published yet. I just published that version now ready for these changes to be merged and published.

**Comment by brycedorn** [comment]
hey, thanks for the quick reply!

I pulled it out into a function per your suggestion - just didn't have any viewbox'd `polyline` or `line`s so didn't think to update those as well. but it's better to keep things consistent 👍 let me know if there's anything else!

**Comment by ConnorAtherton** [comment]
This looks great, thanks for taking the extra time to make this consistent across all the constructors. I'll comment back here after I have published this to npm for you.

**Comment by ConnorAtherton** [comment]
`walkway.js@0.0.8` is up 👍

**Comment by brycedorn** [comment]
thanks for much for the fast turnaround! really appreciated 🐝

**Comment by idlesHand** [comment]
The new update has broken animation for me and on both the demo pages. Already opened issue just thought it might help to get @brycedorn attention too. Don't know how you tested but something key has been missed. Nothing stands out in the code commited.

**Pull request merged**